src/workerd/io/actor-sqlite.c++

```diff
@@ -5,10 +5,28 @@
 #include "actor-sqlite.h"
 #include <algorithm>
 #include <workerd/jsg/exception.h>
+#include <workerd/util/sentry.h>
 #include "io-gate.h"
 
 namespace workerd {
 
+namespace {
+
+// Returns true if a given (set or unset) alarm will fire earlier than another.
+static bool willFireEarlier(kj::Maybe<kj::Date> alarm1, kj::Maybe<kj::Date> alarm2) {
+  KJ_IF_SOME(t2, alarm2) {
```

Comment:
I was thinking that, too -- it's certainly a lot simpler -- but I wasn't sure if there would be weird interaction if someone actually tried to set a `kj::maxValue` alarm... i.e. missing an update such that `getAlarm()` would return a non-null value when null was expected or vice versa.

But there's otherwise little behaviorial difference between an unset alarm and an alarm at infinity, and I guess these comparisons currently only affect the scheduled alarm value, which users can't currently read.  So maybe the simpler version is fine.

Reply:
Changed to simpler version in 784a5536 .